src/openassetio-core/src/utils/Regex.hpp

```diff
@@ -41,6 +41,14 @@ class Regex {
    public:
     explicit Match(const pcre2_code* code);
 
+    ~Match() = default;
+    Match(const Match&) = delete;
```

Comment:
As there's a custom constructor, should the default ctor be deleted?

Reply:
Same question to `Regex` below.

Reply:
DeathwatchedSimpleCppType too?

Reply:
The default constructor is implicitly deleted in these cases, I believe.

From [cppreference](https://en.cppreference.com/w/cpp/language/default_constructor)

> If **there is no user-declared constructor** or constructor template for a class type, the compiler will implicitly declare a default constructor as an inline public member of its class. 

To double check: https://godbolt.org/z/n1o6dG4jc

Do you still think its worth explicitly `delete`ing to be explicit? cppcoreguidelines doesn't seem to mind (at least, the Clang-Tidy checks don't).

Reply:
Having done more reading on it, and looking at your Compiler Explorer link, I'm satisfied it's deleted.

I was merely wondering if it made more sense to be explicit, but seems C++ handles this for us.